server.js

```diff
@@ -1,30 +1,111 @@
-import express from "express";
-import cors from "cors";
-
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// import avocadoSalesData from "./data/avocado-sales.json";
-// import booksData from "./data/books.json";
-// import goldenGlobesData from "./data/golden-globes.json";
-// import netflixData from "./data/netflix-titles.json";
-// import topMusicData from "./data/top-music.json";
-
-// Defines the port the app will run on. Defaults to 8080, but can be overridden
-// when starting the server. Example command to overwrite PORT env variable value:
+import express from "express"
+import cors from "cors"
+
+import books from "./data/books.json"
+
+// Defines the port the app will run on. Defaults to 8080
 // PORT=9000 npm start
-const port = process.env.PORT || 8080;
-const app = express();
+const port = process.env.PORT || 8080
+const app = express()
 
 // Add middlewares to enable cors and json body parsing
-app.use(cors());
-app.use(express.json());
+app.use(cors())
+app.use(express.json())
 
-// Start defining your routes here
+
+// --------------- ROUTES ---------------- //
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
-});
+
+  const LandingPage = {
+    Welcome:
+    "This is an open API with 500 book reviews.",
+    Routes: [
+      {
+          "/books": "All the books",
+          "/books/:title": "The specific title of the book",
+          "/books/top-rated": "How the books are rated, from highest to lowest rated books",
+          "/books/book/:bookID": "The specific bookID",
+          "/random-book": "A random book",
+      }
+    ]
+  }
+  res.send(LandingPage)
+})
+
+// All the books
+app.get('/books', (req, res) => {
+res.status(200).json({
+  data: books,
+  success: true,
+})
+})
+
+// Titles
+app.get('/books/:title', (req, res) => {
+  const { title } = req.params
+
+  const booksTitle = books.filter(
+    (books) => books.title.toLowerCase() === title.toLowerCase())
+
+    if (!booksTitle) {
+      res.status(404).json({
+        data: "Not found",
+        success: false,
+      })
+    } else {
+      res.status(200).json({
+        data: booksTitle,
+        success: true,
+    })
+    }
+})
+
+// Res: shows top rated to lowest rated
+// Using slice() to show top 10 out of all rated books
+app.get('/books/top-rated', (req, res) => {
+  const bookRating = books.sort(
+    (a, b) => b.average_rating - a.average_rating
+  )
+  res.json(bookRating.slice(0, 50))
```

Comment:
This one does not seem to work. Maybe it has to do with the numbers in the slice?